pkg/cloudprovider/aws/instance.go

```diff
@@ -104,31 +104,28 @@ func (p *InstanceProvider) Create(ctx context.Context,
 	return createFleetOutput.Instances[0].InstanceIds[0], nil
 }
 
-func (p *InstanceProvider) Terminate(ctx context.Context, nodes []*v1.Node) error {
-	if len(nodes) == 0 {
+func (p *InstanceProvider) Terminate(ctx context.Context, node *v1.Node) error {
+	if node == nil {
 		return nil
 	}
-	ids := p.getInstanceIDs(nodes)
-
-	_, err := p.ec2api.TerminateInstancesWithContext(ctx, &ec2.TerminateInstancesInput{
-		InstanceIds: ids,
+	id, err := p.getInstanceID(node)
+	if err != nil {
+		return fmt.Errorf("getting instance ID for node %s, %w", node.Name, err)
```

Comment:
If the instance no longer exists, the node controller in CCM will eventually realize the node has no instance to link to, then attempt to terminate the node. Since this node will have a finalizer, we will see a `deletionTimestamp` happen, thus resulting in the termination workflow.

The only way this would happen forever is if the node wanted to be terminated, but had some improperly formed instanceID. In this case, the only viable way to fix this is to format the instanceID properly, or in the case where the instance is already terminated, manually removing the finalizer.